tests/check_remote_addr.c

```diff
@@ -77,16 +77,117 @@ START_TEST (test_remote_addrs)
+    matched_route = r3_tree_match_route(n, entry);
+    ck_assert(matched_route == NULL);
+
     r3_tree_free(n);
 }
 END_TEST
 
 
+void parse_ipv6(const char *ipv6, int nmask, R3Route * route)
+{
+    struct in6_addr    addr6;
+    int ret = inet_pton(AF_INET6, ipv6, (void *)&addr6);
+    ck_assert(ret == 1);
+
+    for (int i = 0; i < 4; i++) {
+        route->remote_addr_v6[i] = ntohl(addr6.__in6_u.__u6_addr32[i]);
+
+        if (nmask >= 32) {
+            route->remote_addr_v6_bits[i] = 32;
+        } else if (nmask > 0) {
+            route->remote_addr_v6_bits[i] = nmask;
+        } else {
+            route->remote_addr_v6_bits[i] = 0;
+        }
+
+        nmask -= 32;
+    }
+}
+
+
+START_TEST (test_remote_addrs_ipv6)
+{
+    R3Node * n      = r3_tree_create(10);
+    R3Route * route  = NULL;
+    match_entry * entry;
+    R3Route *matched_route;
+
+    char * uri0 = "/foo";
+    route = r3_tree_insert_routel(n, 0, uri0, strlen(uri0), &uri0);
+    parse_ipv6("fe80:fe80::1", 128, route);  // "fe80:fe80::1"
+
+    char * uri1 = "/bar";
+    route = r3_tree_insert_routel(n, 0, uri1, strlen(uri1), &uri1);
+    parse_ipv6("fe80:fe80::", 32, route);  // "fe80:fe80::/32"
+
+    char * uri2 = "/goo";
+    route = r3_tree_insert_routel(n, 0, uri2, strlen(uri2), &uri2);
+    parse_ipv6("::1", 128, route);  // "::1"
+
+    char * err = NULL;
+    r3_tree_compile(n, &err);
+    ck_assert(err == NULL);
+
+    entry = match_entry_create("/foo");
+    entry->remote_addr.base = "fe80:fe80::1";
+    entry->remote_addr.len = sizeof("fe80:fe80::1") - 1;
+    matched_route = r3_tree_match_route(n, entry);
+    ck_assert(matched_route != NULL);
+    ck_assert(matched_route->data == &uri0);
+
+    entry = match_entry_create("/foo");
+    entry->remote_addr.base = "fe80:fe80::2";
+    entry->remote_addr.len = sizeof("fe80:fe80::2") - 1;
+    matched_route = r3_tree_match_route(n, entry);
+    ck_assert(matched_route == NULL);
+
+    entry = match_entry_create("/foo");
+    entry->remote_addr.base = "fe88:fe80::1";
+    entry->remote_addr.len = sizeof("fe88:fe80::1") - 1;
+    matched_route = r3_tree_match_route(n, entry);
+    ck_assert(matched_route == NULL);
+
+    entry = match_entry_create("/bar");
+    entry->remote_addr.base = "fe80:fe80::1";
+    entry->remote_addr.len = sizeof("fe80:fe80::1") - 1;
+    matched_route = r3_tree_match_route(n, entry);
+    ck_assert(matched_route != NULL);
+    ck_assert(matched_route->data == &uri1);
+
+    entry = match_entry_create("/bar");
+    entry->remote_addr.base = "fe88:fe80::1";
+    entry->remote_addr.len = sizeof("fe88:fe80::1") - 1;
+    matched_route = r3_tree_match_route(n, entry);
+    ck_assert(matched_route == NULL);
+
+    entry = match_entry_create("/goo");
+    entry->remote_addr.base = "::1";
+    entry->remote_addr.len = sizeof("::1") - 1;
+    matched_route = r3_tree_match_route(n, entry);
+    ck_assert(matched_route != NULL);
+    ck_assert(matched_route->data == &uri2);
```

Comment:
please try to separate test cases if you're testing different test cases.

Reply:
Sorry, I don't understand your question. The current test case for ipv6 has been summarized into `test_remote_addrs_ipv6`.

Or do you think this part of the use case should be written in an independent test file?